src/components/TokenSelect/TokenOptions.tsx

```diff
@@ -148,8 +148,11 @@ const TokenOptions = forwardRef<TokenOptionsHandle, TokenOptionsProps>(function
   ref
 ) {
   const [focused, setFocused] = useState(false)
-  const [selected, setSelected] = useState<Currency>()
-  const hover = useMemo(() => (selected ? tokens.indexOf(selected) : -1), [selected, tokens])
+  const [selected, setSelected] = useState<Currency>(tokens[0])
+  useEffect(() => {
+    setSelected((selected) => (tokens.includes(selected) ? selected : tokens[0]))
+  }, [tokens, setSelected])
```

Comment:
in practice tokens will never change between renders right? so this is only run on load

Reply:
no, this will run when a user modifies the `tokens` by entering search terms.
eg when a user enters `UNI` `tokens` will be re-passed as `[<UNI>]` and the component will re-render.

Effectively this is a guard that says "if there is no selected token when the user searches, select the one at the top of the list".

Reply:
ah! i would definitely explain that in code somehow. not obvious. either a comment or name some variables